x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/optimizer/OptimizerRunTests.java

```diff
@@ -48,4 +57,57 @@ public void testWhereClause() {
         LogicalPlan p = plan("SELECT some.string l FROM test WHERE int IS NOT NULL AND int < 10005 ORDER BY int");
         assertNotNull(p);
     }
-}
\ No newline at end of file
+
+    public void testReducedBinaryComparisonGreaterThenOrEqual() {
+        // i >= 12
+        doTestBinaryComparisonReduction("((int + 1) / 2 - 3) * 4 >= 14", GreaterThanOrEqual.class, 12d);
+    }
+
+    public void testReducedBinaryComYparisonLessThen() {
+        // i < -5/6
+        doTestBinaryComparisonReduction("12 * (-int / 5) > (8 + 12) / 10", LessThan.class, -5d / 6);
+    }
+
+    public void testReducedBinaryComYparisonNotEquals() {
+        // i != 7000
+        doTestBinaryComparisonReduction("-3600 != (int - 200) / 2", NotEquals.class, -7000);
+    }
+
+    public void testReducedBinaryComparisonEquals() {
+        // i = -12
+        doTestBinaryComparisonReduction("2 * 3 / (4 / -int) = 18", Equals.class, -12d);
+    }
+
+    public void testReducedBinaryComparisonWithConjunction() {
+        doTestBinaryComparisonReduction("2 * 3 / (4 / -int) = 18 AND int >= -12", Equals.class, -12d);
+    }
+
+    public void testReducedBinaryComparisonWithDisjunction() {
+        doTestBinaryComparisonReduction("2 * 3 / (4 / -int) = 18 OR int > -12", GreaterThanOrEqual.class, -12d);
+    }
+
+    private void doTestBinaryComparisonReduction(String expression, Class<? extends BinaryComparison> binaryComparisonClass,
+                                                 Number bound) {
+        LogicalPlan plan = plan("SELECT some.string FROM test WHERE " + expression);
+
+        assertTrue(plan instanceof UnaryPlan);
+        UnaryPlan unaryPlan = (UnaryPlan) plan;
+        assertTrue(unaryPlan.child() instanceof Filter);
+        Filter filter = (Filter) unaryPlan.child();
+        assertEquals(binaryComparisonClass, filter.condition().getClass());
+        BinaryComparison bc = (BinaryComparison) filter.condition();
+
+        assertTrue(bc.left() instanceof FieldAttribute);
+        FieldAttribute attribute = (FieldAttribute) bc.left();
+        assertEquals("int", attribute.name());
+
+        assertTrue(bc.right() instanceof Literal);
```

Comment:
The test only cares about the expression so you can just use `parser.createExpression` (see tests inside `ExpressionTests`).

Reply:
I've employed `parser.createExpression()`, thanks.